### PR TITLE
Show warning when user attempts to disable PR edits

### LINKED ISSUE #1665: Show warning if PR submitter tries to uncheck `Allow edits from maintainers`
I often get PRs where the user has unchecked `Allow edits from maintainers.` when creating a PR. This is super annoying as it means I cannot easily do changes to the PR myself. There's no reason for users to do this. Would be nice to show a warning if a user tries to uncheck it that explains why unchecking it would be annoying to the maintainers.

screen shot 2018-12-12 at 21 19 07

### PR DESCRIPTION
**Pull request by CEOehis**
closes #1665

### CODE REVIEW

File: source/features/allow-edits-unchecked-warning.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	if (warning) {
+		warning.remove();
+	}
+}
+
+function init() {
+	const checkBox = select<HTMLInputElement>('.js-collab-option');
+
+	checkBox.addEventListener('change', function() {
+		if(!checkBox.checked) {
```
Comment (bfred-it) [inline source/features/allow-edits-unchecked-warning.tsx]: Change this to: `if checked remove, else append`

File: source/features/allow-edits-unchecked-warning.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+function removeWarning() {
+	const warning = select('#allow-edits-unchecked-warning');
+	if (warning) {
+		warning.remove();
+	}
+}
+
+function init() {
+	const checkBox = select<HTMLInputElement>('.js-collab-option');
+
```
Comment (bfred-it) [inline source/features/allow-edits-unchecked-warning.tsx]: What if the message is generated once here (no need for an extra function) and then it’s just appended and removed inside the event listener?

Comment (CEOehis) [reply source/features/allow-edits-unchecked-warning.tsx]: That totally makes sense. Thanks for pointing that out.

File: source/features/allow-edits-unchecked-warning.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	checkBox.addEventListener('change', () => {
+		if (checkBox.checked) {
+			return removeWarning();
+		}
+
+		select('.new-discussion-timeline .composer .timeline-comment').after(
+			<div class="flash flash-error" id="allow-edits-unchecked-warning">
+				<strong>Note:</strong> Disabling this would prevent the maintainer of this project from making vital changes to your branch.
```
Comment (yakov116) [inline source/features/allow-edits-unchecked-warning.tsx]: `the maintainer` => a maintainer

**Comment by CEOehis** [comment]
@bfred-it which is preferred for fixing conflicts, rebasing off of master or merging master into my branch?

File: source/features/allow-edits-unchecked-warning.tsx (hunk 2 above)
Comment (bfred-it) [inline source/features/allow-edits-unchecked-warning.tsx]: This is still outstanding

Comment (bfred-it) [reply source/features/allow-edits-unchecked-warning.tsx]: This is still outstanding

Comment (CEOehis) [reply source/features/allow-edits-unchecked-warning.tsx]: Done. I originally thought you were talking about the `generateWarning` function. I've made the necessary fix.

**Comment by bfred-it** [comment]
Your last force push undid some of my changes. `--force` isn't necessary on PRs since they'll be squashed anyway.

**Comment by CEOehis** [comment]
> I think your last force push undid some of my changes. `--force` isn't necessary on PRs since they'll be squashed anyway.

Noted. The force push was to fix a Travis linting issue though.

File: source/features/allow-edits-unchecked-warning.tsx (hunk 2 above)
Comment (bfred-it) [reply source/features/allow-edits-unchecked-warning.tsx]: c892e92 is how I meant it

**Comment by bfred-it** [comment]
@sindresorhus suggestions for the copy?

**Comment by sindresorhus** [comment]
@bfred-it Something like this?

> Disabling this prevents the maintainers from doing important changes to your pull request which can result in it taking longer to merge or risk it not being merged at all.

**Comment by bfred-it** [comment]
> @bfred-it Something like this?
> 
> > Disabling this prevents the maintainers from doing important changes to your pull request which can result in it taking longer to merge or risk it not being merged at all.

A bit too formal and a single long sentence. How about:

> Maintainers want change. Maintainers can't change. Maintainers be angry. Maintainers block your PR. Don't give anger to maintainers [🐻](https://www.youtube.com/watch?v=AdYaTa_lOf4)

Or:

> Maintainers may require changes. It’s easier and faster to allow them to make direct changes before merging.

**Comment by CEOehis** [comment]
> 
> Or:
> 
> > Maintainers may require changes. It’s easier and faster to allow them to make direct changes before merging.

This one sounds better.

**Comment by sindresorhus** [comment]
> Maintainers may require changes. It’s easier and faster to allow them to make direct changes before merging.

👍

**Comment by bfred-it** [comment]
---

**Comment by bfred-it** [comment]
I'm trying to push but, uhm... your checkbox was unchecked!! @CEOehis 😆

**Comment by CEOehis** [comment]
> @CEOehis I'm trying to push but, uhm... your checkbox was unchecked! It's easier and faster to allow me to make direct changes before merging. 🤣

🤣 	I'm sorry. I forgot to toggle it back when I was testing the feature on the sidebar. I've reenabled edits @bfred-it

**Comment by sindresorhus** [comment]
Looks good 👌

**Comment by busches** [comment]
Getting an error when on any PR, not just opening PRs:
```
content.js:185 TypeError: Cannot read property 'checked' of null
    at update (content.js:6423)
    at init (content.js:6429)
    at run (content.js:180)
    at load (content.js:232)
    at onAjaxedPagesRaw (content.js:137)
    at onAjaxedPagesRaw (content.js:132)
```
We need to check if the checkbox is there and likely shortcut if it doesn't exist.

**Comment by bfred-it** [comment]
Reported in https://github.com/sindresorhus/refined-github/issues/1811

**Pull request merged**